AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -165,26 +181,38 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
             return
         }
 
-        let syncMutationToCloudOperation =
-            SyncMutationToCloudOperation(mutationEvent: mutationEvent, api: api) { [weak self] result in
-                self?.log.verbose(
-                    "[SyncMutationToCloudOperation] mutationEvent finished: \(mutationEvent); result: \(result)")
-
-                if case .completed(let response) = result,
-                    case .failure(let error) = response {
-
-                    let processMutationErrorFromCloudOperation =
-                        ProcessMutationErrorFromCloudOperation(mutationEvent: mutationEvent,
-                                                               error: error) { [weak self] result in
-
-                        self?.log.verbose(
-                            "[ProcessMutationErrorFromCloudOperation] mutation error processed, result: \(result)")
-                        self?.completeProcessingEvent(mutationEvent)
-                    }
-                    self?.operationQueue.addOperation(processMutationErrorFromCloudOperation)
-                } else {
-                    self?.completeProcessingEvent(mutationEvent)
+        let syncMutationToCloudOperation = SyncMutationToCloudOperation(mutationEvent: mutationEvent,
+                                                                        api: api) { [weak self] result in
```

Comment:
Still not entirely convinced we need to do any [weak self] casts here. have we run the memory leak analyzer?

Reply:
i haven't ran much else other than the one you showed me that day that said there were no issues, i think for operations that are short lived, it is stay to keep the [weak self] in case the operation goes out of scope but then doesn't get deinitialized because there's a strong reference to `self` which is the OutgoingMutationQueue

Reply:
My concern is that it sets a precedence for the rest of the code base, and this seems like this should only be added if we know that we are actually running into a case where it's needed.

In this case, I don't think it's needed because the `OutgoingMutationQueue` is expected to be in memory until the system comes down because there is a strong reference from the RemoteSyncEngine that is expected to live for the life that the application is in memory (afaik, OutgoingMutationQueue gets created in init() and then never gets de-allocated, and similarly in StorageEngine, it creates the SyncEngine in init(), and then never deallocates it).  When I see the use of `[weak self]` here, this tells me that we are expecting that the OutgoingMutationQueue will be going out of memory/scope and if that happens the callback from SyncMutationToCloudOperation (and ProcessMutationErrorFromCloudOperation) should not continue w/ the guard statement.  If this is behavior we do want, then great-- but I don't think this is the case.

Reply:
thanks for explaining, removing weak self here since we don't expect OutgoingMutatioinQueue to be deinitialized